src/Configuration/Compiler/PublicServiceVisibilityCompilerPass.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace GrumPHP\Configuration\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Turns default private services into public services via a compiler pass,
+ * instead of modifying all service definition across several files.
+ *
+ * This mitigates the issues where fetching services out of the container in
+ * Symfony 3.4 / 4.0 will trigger an error, because they now get declared as
+ * a "private" service by default.
+ */
+class PublicServiceVisibilityCompilerPass implements CompilerPassInterface
```

Comment:
I would prefer to add the visibility to the config file explicitly instead of marking all services as true.

Reply:
that will be a lot of "changes" then. I was not quite sure I was allowed to make them.

Will look into it this weekend, when I find some time. Currently a bit slammed at work 😃

Reply:
Maybe a [explicit service locators](https://github.com/symfony/symfony/pull/21708) might be a good fit, too? That way, we don't have to inject the whole container (which is considered a bad practice), _and_ we also don't have to declare all of these 'private' services as public, just to please the 4.0 container?